### PR TITLE
pscanrules: Add Example Alert to XAspNetVersionScanRule

### PR DESCRIPTION
**Pull request by thgomes**
Part of https://github.com/zaproxy/zaproxy/issues/6119

Changes:
- Add example alert functionality.
- Add unit tests for example alerts functionality.
- Add note to changelog.

### CODE REVIEW

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRule.java [hunk 1]
```diff
@@ -57,25 +57,18 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
...
+                        found.get(0),
+                        Constant.messages.getString(MESSAGE_PREFIX + "soln"),
+                        Constant.messages.getString(MESSAGE_PREFIX + "desc")
+                )
+                .raise();
             }
         }
     }
 
-    private void raiseAlert(HttpMessage msg, int id, String evidence) {
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRule.java:65]: Do not move the method, also, static values (like description) should be the same for example alerts.

**Comment by thc202** [comment]
This has a conflict now.

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRule.java [hunk 2]
```diff
@@ -100,6 +110,18 @@ public int getCweId() {
     }
 
     public int getWascId() {
-        return 14; //  WASC-14: Server Misconfiguration
+        return 14; // WASC-14: Server Misconfiguration
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRule.java:113]: Revert.

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRule.java [hunk 3]
```diff
@@ -57,23 +57,33 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
...
                 .setRisk(getRisk())
-                .setConfidence(Alert.CONFIDENCE_HIGH)
-                .setDescription(Constant.messages.getString(MESSAGE_PREFIX + "desc"))
+                .setConfidence(confidence)
+                .setDescription(description)
                 .setOtherInfo(Constant.messages.getString(MESSAGE_PREFIX + "extrainfo"))
-                .setSolution(Constant.messages.getString(MESSAGE_PREFIX + "soln"))
-                .setReference(Constant.messages.getString(MESSAGE_PREFIX + "refs"))
+                .setSolution(solution)
+                .setReference(reference)
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRule.java:83]: Revert these.

File: addOns/pscanrules/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 ### Changed
+- The XAspNetVersion Scan Rule now includes example alert functionality for documentation generation purposes (Issue 6119).
```
Comment (thc202) [inline addOns/pscanrules/CHANGELOG.md:8]: It's better to use the exact name `X-AspNet-Version Response Header`.

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRuleUnitTest.java [hunk 5]
```diff
@@ -99,6 +102,25 @@ void shouldNotRaiseAlertWhenResponseDoesNotContainsXAspNetVersionHeader()
...
+        String MESSAGE_PREFIX = "pscanrules.xaspnetversion.";
+
+        List<Alert> alerts = rule.getExampleAlerts();
+
+        Alert alert = alerts.get(0);
+        assertThat(alert.getConfidence(), equalTo(Alert.CONFIDENCE_HIGH));
+        assertThat(
+                alert.getReference(),
+                equalTo(Constant.messages.getString(MESSAGE_PREFIX + "refs")));
+        assertThat(alert.getEvidence(), equalTo("Example evidence"));
```
Comment (kingthorin) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/XAspNetVersionScanRuleUnitTest.java:116]: Should use an actual example.

**Comment by kingthorin** [comment]
@thgomes are you going to finish this?

**Comment by thgomes** [comment]
> @thgomes are you going to finish this?

Sorry for the delay, I'm uploading a correction.

**Review by psiinon — APPROVED** [review]

**Comment by thc202** [comment]
@thgomes how would you like to be credited (e.g. name, handle)?
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

**Pull request merged**